ext/srt/gstsrtobject.c

```diff
@@ -1531,12 +1531,16 @@ gst_srt_object_write_one (GstSRTObject * srtobject,
       case SRTS_BROKEN:
       case SRTS_NONEXIST:
       case SRTS_CLOSED:
-        GST_WARNING_OBJECT (srtobject->element,
-            "Invalid SRT socket. Trying to reconnect");
+        GST_ELEMENT_WARNING (srtobject->element, RESOURCE, WRITE, NULL,
+            ("Invalid SRT socket. Trying to reconnect. (%s)",
+                srt_getlasterror_str ()));
         gst_srt_object_close (srtobject);
         if (!gst_srt_object_open_internal (srtobject, cancellable, error)) {
           return -1;
         }
+        /* We can't send into a broken SRT socket. This will drop the current
```

Comment:
I'm not sure if silently dropping buffers while we attempt to repair the broken connection is the best thing to do. Perhaps we should cease trying to reconnect, treat invalid socket as an unrecoverable state, return `GST_FLOW_ERROR` from `gst_srt_sink_render()` and stop streaming?